### PR TITLE
Update ext-locale-init.reb

### PR DESCRIPTION
**Pull request by gchiu**
changed so that returns a default US English if it can't find or parse the language and locale.  And no longer errors but prints message.

### CODE REVIEW

File: src/extensions/locale/ext-locale-init.reb [hunk 1]
```diff
@@ -453,34 +454,40 @@ unless 'Windows = first system/platform [
...
             ]
             find? [territory territory*] type [
-                select iso-3166 territory
+                unless territory: select iso-3166 territory [print error-msg]
+                territory: default "US"
             ]
             true [
-                fail spaced ["Invalid locale type:" type]
+                fail unspaced ["Unrecognised env LANG value" env-lang]
             ]
```
Comment (zsx) [inline src/extensions/locale/ext-locale-init.reb]: I don't think this change is correct: it was doing the input validation.

File: src/extensions/locale/ext-locale-init.reb [hunk 2]
```diff
@@ -453,34 +454,40 @@ unless 'Windows = first system/platform [
...
         letter: charset [#"a" - #"z" #"A" - #"Z"]
         unless parse env-lang [
             copy lang: [some letter]
-            [#"_" copy territory: [some letter] | #"." (territory: copy "US")]
-            to end (if lang = "C" [lang: copy "en"])
+            #"_" copy territory: [some letter]
+            to end
         ][
-            fail spaced ["Non-standard env LANG:" env-lang]
+            if lang <> "C" [print error-msg]
```
Comment (zsx) [inline src/extensions/locale/ext-locale-init.reb]: I believe "POSIX" is also a valid setting that's not in the ISO tables: http://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap08.html

**Comment by draegtun** [comment]
For reference this is how Perl handles a malformed LANG (locale):

    $ LANG=C.UTF-8 perl -E 'say "HERE"'
    perl: warning: Setting locale failed.
    perl: warning: Please check that your locale settings:
        LC_ALL = (unset),
        LANG = "C.UTF-8"
        are supported and installed on your system.
    perl: warning: Falling back to the standard locale ("C").
    HERE

So doesn't fail but produces a warning and sets locale to `C` (default UNIX locale).

I believe Perl uses [`POSIX::setlocale`](https://perldoc.perl.org/perllocale.html#The-setlocale-function) to set/check the locale.  @zsx - Are (or can?) POSIX functions like this available to use in Rebol user-space @zsx ?

File: src/extensions/locale/ext-locale-init.reb (hunk 2 above)
Comment (draegtun) [inline src/extensions/locale/ext-locale-init.reb]: @zsx - Yep POSIX is a valid setting...

    $ LANG=POSIX perl -E 'say "HERE"'
    HERE

File: src/extensions/locale/ext-locale-init.reb (hunk 1 above)
Comment (gchiu) [inline src/extensions/locale/ext-locale-init.reb]: Well, it shouldn't get here unless it's a really malformed LANG

File: src/extensions/locale/ext-locale-init.reb (hunk 2 above)
Comment (gchiu) [inline src/extensions/locale/ext-locale-init.reb]: Does anyone want to have a try at this?

**Comment by zsx** [comment]
@draegtun No, currently user space has no access to `setlocale`, unless through FFI/user natives.

File: src/extensions/locale/ext-locale-init.reb (hunk 2 above)
Comment (draegtun) [inline src/extensions/locale/ext-locale-init.reb]: @gchiu - Here's what I get on Mac OSX (Unix)...

    $ LANG=C.UTF-8 ./r3 --do "print {HERE}"
    HERE

    $ LANG=POSIX ./r3 --do "print {HERE}"
    *** Unrecognised env LANG: POSIX  - defaulting to US English ***
    *** Unrecognised env LANG: POSIX  - defaulting to US English ***
    *** Unrecognised env LANG: POSIX  - defaulting to US English ***
    *** Unrecognised env LANG: POSIX  - defaulting to US English ***
    HERE
    

So not failing but getting 4 warning messages on POSIX.

Comment (gchiu) [inline src/extensions/locale/ext-locale-init.reb]: I think that's because in start up locale is called 4 times.

**Comment by gchiu** [comment]
So, what's the best way to handle this?  There's a big warning not to alter the tables as they are sourced elsewhere.  Should we add conditions for things not in the ISO tables such as _POSIX_ ?

**Comment by zsx** [comment]
The issue this pull tries to fix should have been fixed by https://github.com/metaeducation/ren-c/commit/ce53743475915fb17e9e1442e60d8f7b4de23cb6, so closing.

**Comment by hostilefork** [comment]
Cool... @zsx, could you also tie up and delete any resolved branches on the master?  (ext-locale, etc.)

**Pull request closed without merging**